### DISCUSSION TITLE
Artificially Terminate Function

### DISCUSSION THREAD

**Discussion by endeav0r** [Vector35/binaryninja-api#2603] [General] [answered]
I have a function which makes a system call. That system call will never return. There is data after that system call I would like to have XREFs too. Is there a way I can say, "This function does not continue beyond this instruction," allowing me to typ-ify the bytes following the system call?

**Comment by psifertex** [comment, marked as answer]
So one answer is with an arch hook. 

The other (easier) answer is to create syscall for that platform that defines the type and sets it as a no-return. 

Create a file in `~/.binaryninja/types/platform/$plat-$arch.c` that contains something like the following:

```
void sys_exit(int status) __noreturn __syscall(93);
void sys_exit_group(int status) __noreturn __syscall(94);
```

I _think_ that should get you what you want?
